### PR TITLE
Show notice if a non-prod site may be indexable

### PR DESCRIPTION
**Pull request by trepmal**
Creates an admin notice indicating the site may be indexable if `blog_public` option is `1`

Screen Shot 2020-05-15 at 2 59 44 PM

Additionally, does not show a notice if:
 - the current user does not have `manage_options`
 - the notice has been dismissed (saved in options, reset if blog_public is resaved to `1`)
 - the home url ends in `.go-vip.co` or `.go-vip.net`

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Example:

1. Check out PR and/or otherwise run this file as an mu-plugin. Only dependency is `wp_endswith()` [from here](https://github.com/Automattic/vip-go-mu-plugins/blob/ef9d3ca1038275e7abe5f39fcbd12410506c4d04/vip-helpers/vip-utils.php#L348) which, despite the prefix, is in fact not a core function.
1. If needed, define `VIP_GO_APP_ENVIRONMENT` as anything but `production`
1. If you do not immediately see a notice in WP-Admin, blog_public may already be `0`, in Reading Options, toggle and resave the Search Engine Visibility  checkbox.
1. With blog_public set to `1`, the notice should be visible

Things to check:
1. clicking the x should dismiss the notice, and it _should not_ return on refresh
1. toggle and save the Search Engine Visibility setting, notice should be restored
1. change `VIP_GO_APP_ENVIRONMENT` to `production`, notice should be gone
1. check on a go-vip.co/net site, notice should not appear

### CODE REVIEW

**Comment by trepmal** [comment]
Retesting on Go sandbox. dismissal flag seems to be misbehaving there, but works on my local. May be interference with another plugin

**Comment by trepmal** [comment]
> Retesting on Go sandbox. dismissal flag seems to be misbehaving there, but works on my local. May be interference with another plugin

 type issue. 1f4901908fa2cb84d944c649bb1ac03291bd9032 address this. Currently unsure of what causes the difference between sandbox and my local.

File: blogpublic-notice.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ */
```
Comment (emrikol) [inline blogpublic-notice.php:3]: PHPCS: Doc comment is empty
PHPCS: Missing @Package tag in file comment

File: blogpublic-notice.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace Automattic\VIP\Blog_Public;
+
+if ( 'production' === VIP_GO_APP_ENVIRONMENT ) {
+	return;
+}
+
+/**
```
Comment (emrikol) [inline blogpublic-notice.php:11]: PHPCS: Doc comment for parameter "$wp_admin_bar" missing

File: blogpublic-notice.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+function dismiss_callback() {
+	if ( current_user_can( 'manage_options' ) ) {
+		update_option( 'blogpublic_notice_dismissed', true, 'no' );
+		wp_send_json_success();
+	}
+	wp_send_json_error();
+}
+add_action( 'wp_ajax_blogpublic_notice_dismiss', __NAMESPACE__ . '\dismiss_callback' );
+
+/**
```
Comment (emrikol) [inline blogpublic-notice.php:69]: PHPCS: Doc comment for parameter "$old_value" missing
PHPCS: Doc comment for parameter "$value" missing
PHPCS: Doc comment for parameter "$option" missing

**Review by emrikol — DISMISSED** [review]

**Comment by emrikol** [comment]
> type issue. [1f49019](https://github.com/Automattic/vip-go-mu-plugins/commit/1f4901908fa2cb84d944c649bb1ac03291bd9032) address this. Currently unsure of what causes the difference between sandbox and my local.

IIRC it's because we store options as strings (because memcache maybe?) on VIP Go and WPCOM

**Review by rinatkhaziev — APPROVED** [review]
Thank you @trepmal

File: blogpublic-notice.php (hunk 1 above)
Comment (brandon-m-skinner) [inline blogpublic-notice.php:3]:
```suggestion
```

**Review by brandon-m-skinner — APPROVED** [review]

**Comment by brandon-m-skinner** [comment]
r1394-stacks

**Pull request merged**